controllers/controllers/workloads/cforg_controller_test.go

```diff
@@ -0,0 +1,415 @@
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/hierarchical-namespaces/api/v1alpha2"
+)
+
+var _ = Describe("CFOrgReconciler", func() {
+	var (
+		fakeClient       *fake.Client
+		fakeStatusWriter *fake.StatusWriter
+
+		cfOrgGUID string
+
+		cfOrg              *workloadsv1alpha1.CFOrg
+		subNamespaceAnchor *v1alpha2.SubnamespaceAnchor
+		namespace          *v1.Namespace
+
+		setSubNamespaceStatus   bool
+		cfOrgError              error
+		subNamespaceAnchorError error
+		createError             error
+		namespaceError          error
+
+		cfOrgReconciler *CFOrgReconciler
+		ctx             context.Context
+		req             ctrl.Request
+
+		reconcileResult ctrl.Result
+		reconcileErr    error
+	)
+
+	BeforeEach(func() {
+		fakeClient = new(fake.Client)
+
+		cfOrgGUID = PrefixedGUID("cf-org")
+
+		cfOrg = BuildCFOrgObject(cfOrgGUID, defaultNamespace)
+		subNamespaceAnchor = BuildSubNamespaceAnchorObject(cfOrgGUID, defaultNamespace)
+		namespace = BuildNamespaceObject(cfOrgGUID)
+
+		setSubNamespaceStatus = true
+		cfOrgError = nil
+		subNamespaceAnchorError = nil
+		namespaceError = nil
+		createError = nil
+		reconcileErr = nil
+
+		fakeClient.GetStub = func(_ context.Context, _ types.NamespacedName, obj client.Object) error {
+			// cast obj to find its kind
+			switch obj := obj.(type) {
+			case *workloadsv1alpha1.CFOrg:
+				cfOrg.DeepCopyInto(obj)
+				return cfOrgError
+			case *v1alpha2.SubnamespaceAnchor:
+				subNamespaceAnchor.DeepCopyInto(obj)
+				return subNamespaceAnchorError
+			case *v1.Namespace:
+				namespace.DeepCopyInto(obj)
+				return namespaceError
+			default:
+				panic("TestClient Get provided a weird obj")
+			}
+		}
+
+		fakeClient.CreateStub = func(ctx context.Context, obj client.Object, option ...client.CreateOption) error {
+			// cast obj to find its kind
+			switch obj := obj.(type) {
+			case *v1alpha2.SubnamespaceAnchor:
+				if setSubNamespaceStatus {
+					obj.Status.State = v1alpha2.Ok
+				}
+				return createError
+			default:
+				panic("TestClient Create provided an unexpected object type")
+			}
+		}
+
+		// Configure mock status update to succeed
+		fakeStatusWriter = &fake.StatusWriter{}
+		fakeClient.StatusReturns(fakeStatusWriter)
+
+		// configure a CFOrgReconciler with the client
+		Expect(workloadsv1alpha1.AddToScheme(scheme.Scheme)).To(Succeed())
+		cfOrgReconciler = &CFOrgReconciler{
+			Client: fakeClient,
+			Scheme: scheme.Scheme,
+			Log:    zap.New(zap.WriteTo(GinkgoWriter), zap.UseDevMode(true)),
+		}
+		ctx = context.Background()
+		req = ctrl.Request{
+			NamespacedName: types.NamespacedName{
+				Namespace: defaultNamespace,
+				Name:      cfOrgGUID,
+			},
+		}
+	})
+
+	JustBeforeEach(func() {
+		reconcileResult, reconcileErr = cfOrgReconciler.Reconcile(ctx, req)
+	})
+
+	When("a CFOrg is created and CFOrgReconciler Reconcile function is called", func() {
+		BeforeEach(func() {
+			cfOrg.DeletionTimestamp = nil
+		})
+
+		When("on the happy path", func() {
```

Comment:
This `When` does not add any information and can be omitted

Reply:
Done